content/kotlin/concepts/functions/functions.md

```diff
@@ -0,0 +1,200 @@
+---
+Title: 'Functions'
+Description: 'Functions are the most essential part of kotlin. A function is a reusable block of code that can be called from elsewhere in the program.'
+Subjects: 
+  - 'Mobile Development'
+Tags: 
+  - 'Android'
+  - 'Functions'
+  - 'Kotlin'
+CatalogContent: 
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+
+**Functions** are an essential part of any Kotlin program. Functions are a reusable part of code that can be called in any part of the program. The Kotlin program has to start with the `main()` function, or there will be an error. `fun` is used to define a function in Kotlin, but there are also predefined functions like `println `and `print`.
+
+## Function `main()`
+
+Function `main()` is the entry point to a Kotlin program and is required. The `main()` function is meant to include the code of the program and is not called anywhere. The code should be between the `main()` function curly brackets `{}`.
+
+```
+fun main() {
+  println("Hello World")
+}
+```
+
+## Creating Functions in Kotlin
+
+User defined functions start with the `fun` keyword, followed by the function name, the parameters (if there are any), and the function body which is enclosed in curly braces.
+
+```pseudo
+fun functionName(arguments): return type {
+  body
+  return statment
+}
+```
+
+- `fun` is used to define a function.
+- The `functionName` is used to call the function in the program.
+- `Arguments` have to be in the parentheses. Arguments are pieces of information fed to the function.
+- `Return type` declares the type of a returning value. It is optional.
+- The `body` contains the code to be executed.
+- `Return statement` returns the value of the function.
+
+### Arguments
+
+Arguments are data passed to the function. They must have a name and data type. Between the argument name and data type must be a colon `:`. The arguments are separated by comma `,` .
+
+```pseudo
+fun functionName(argumentName: dataType, argumentName2 : datatype){
+  body
+}
+```
+
+```pseudo
+fun sample(num1: Int,num2: Int){
+  println(num1 + num2)
+}
+```
+
+When calling the function, `arguments` can be put in their assigned spot or specified by name. By specifying the name the `arguments` do not need to be in the same order as in the function header.
+
+```pseudo
+fun main(){
+  println(sample(3,5))
+}
+fun main(){
```

Comment:
Make this two code blocks so there's not two `main` functions in one code block.